public/app/plugins/datasource/loki/variables.ts

```diff
@@ -0,0 +1,30 @@
+import { from, Observable } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+
+import { CustomVariableSupport, DataQueryRequest, DataQueryResponse } from '@grafana/data';
+
+import { LokiVariableQueryEditor } from './components/VariableQueryEditor';
+import { LokiDatasource } from './datasource';
+import { LokiMetricFindQuery } from './metricFindQuery';
+import { LokiVariableQuery } from './types';
+
+export class LokiVariableSupport extends CustomVariableSupport<LokiDatasource, LokiVariableQuery> {
+  private metricFindQuery: LokiMetricFindQuery;
+
+  constructor(datasource: LokiDatasource) {
+    super();
+    this.metricFindQuery = new LokiMetricFindQuery(datasource);
```

Comment:
Just following with what was done in `cloud-monitoring` https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/cloud-monitoring/variables.ts#L19 . It could very well be another class method.

Reply:
I assume they used another class because of their multiple query types. Until now it would only be two types for Loki, which would have been manageable in the `variables.ts` file itself. Nevertheless, thanks for the explanation.

Reply:
Yeah, it's more complex than hours. I would personally prefer not to use this new class at all.

Reply:
Totally your choice. I would also be happier without the extra class.

Reply:
Removed!